test/splitOnRegex.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+var jsc = require('jsverify');
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+var equals = require('./internal/equals');
+
+
+test('splitOnRegex', function() {
+
+  eq(typeof S.splitOnRegex, 'function');
+  eq(S.splitOnRegex.length, 2);
+  eq(S.splitOnRegex.toString(), 'splitOnRegex :: RegExp -> String -> Array String');
+
+  eq(S.splitOnRegex(/b/, 'abc'), ['a', 'c']);
+  eq(S.splitOnRegex(/d/, 'abc'), ['abc']);
+  eq(S.splitOnRegex(/b(?=c)/, 'abc abd'), ['a', 'c abd']);
+  eq(S.splitOnRegex(/b(?!c)/, 'abc abd'), ['abc a', 'd']);
+
+  eq(S.splitOnRegex(/\s+/, 'foo  bar baz   qux'), ['foo', 'bar', 'baz', 'qux']);
+  eq(S.splitOnRegex(/\s+/, 'foo  bar baz\tqux\nquux'), ['foo', 'bar', 'baz', 'qux', 'quux']);
+  eq(S.splitOnRegex(/\s+/, 'foobar'), ['foobar']);
+  eq(S.splitOnRegex(/;(;)?/, 'foo;;bar;baz'), ['foo', 'bar', 'baz']);
+
+  eq(S.splitOnRegex(/./, 'a\rb\nc'), ['', '\r', '\n', '']);
+  eq(S.splitOnRegex(/[^]/, 'a\nb'), ['', '', '', '']);
+
+  eq(S.splitOnRegex(/^.*$/m, 'foo\nbar'), ['', '\n', '']);
+  eq(S.splitOnRegex(/^.*$/, 'foo\nbar'), ['foo\nbar']);
+
+  eq(S.splitOnRegex(/foo/, 'FOObar'), ['FOObar']);
+  eq(S.splitOnRegex(/foo/i, 'FOObar'), ['', 'bar']);
+
+  eq(S.splitOnRegex(/b/, 'abc'), ['a', 'c']);
+  eq(S.splitOnRegex(/(?:b)/, 'abc'), ['a', 'c']);
+
+  eq(S.splitOnRegex(/(?:)/, ''), []);
+  eq(S.splitOnRegex(/(?:)/, 'hello'), ['h', 'e', 'l', 'l', 'o']);
+  eq(S.splitOnRegex(/.*/, ''), []);
+  eq(S.splitOnRegex(/.*/, 'hello'), ['']);
+  eq(S.splitOnRegex(/./, ''), ['']);
+  eq(S.splitOnRegex(/./, 'hello'), ['', '', '', '', '', '']);
+
+  jsc.assert(jsc.forall(jsc.nestring, function(s) {
+    return equals(S.splitOnRegex(S.regex('', S.regexEscape(s)), s))(['', ''])
+        && equals(S.splitOnRegex(/[^]/, s).length)(s.length + 1)
+        && equals(S.splitOnRegex(/(?:)/, s))(S.splitOn('', s));
+  }), {tests: 1000});
+
+  jsc.assert(jsc.forall(jsc.asciistring, function(t) {
+    var min = 0;
+    var max = t.length;
+    var i = jsc.random(min, max);
+    var j = jsc.random(min, max);
+    var s = t.slice(Math.min(i, j), Math.max(i, j));
+    return S.joinWith(s, S.splitOnRegex(S.regex('', S.regexEscape(s)), t)) === t;
+  }), {tests: 1000});
```

Comment:
Nice! Let's also add the property to the description using the same format we use for `joinWith`. :)